crates/control/src/controllers/connector_images.rs

```diff
@@ -1,18 +1,24 @@
-use axum::extract::Extension;
+use axum::extract::{Extension, Path};
 use axum::response::IntoResponse;
 use axum::Json;
 use hyper::StatusCode;
 use sqlx::PgPool;
 
-use crate::controllers::Payload;
+use crate::controllers::json_api::RawJson;
 use crate::error::AppError;
 use crate::models::connector_images::CreateConnectorImage;
+use crate::models::Id;
 use crate::repo::connector_images as images_repo;
+use crate::services::connectors;
+use crate::services::subprocess::Subprocess;
+
+pub mod routes;
```

Comment:
Yeah, `links` is a definitely better name for what I've got here. In Rails/Phoenix, these route helpers *are* generated by the central Router. This was kind of a stopgap since I didn't have a plan for doing that.

However, things move quickly. *After* I added these `routes` modules, a plan for [Type safe routing](https://github.com/tokio-rs/axum/pull/756) was formulated and merged into Axum. This morning they released a new version that contains this work. I'll be revisiting all the route/link generation stuff as part of that upgrade.

Reply:
neat! Yeah, at this point it makes sense to just look into switching to the type safe routing instead of fiddling with that ourselves.